### PR TITLE
Document build procedure

### PR DESCRIPTION
**Pull request by Arcanemagus**
This PR adds the necessary `devDependencies`, and documents the build procedure for the transpiled code.

@nmote this is what we were talking about.

[Review on Reviewable](https://reviewable.io/reviews/atomlinter/linter-eslint/412)

### CODE REVIEW

**Comment by Arcanemagus** [comment]
Did I miss anything @steelbrain?

My only concern here is that `npm install` will also attempt to run `ucompile go`, but I don't _think_ that `apm` will run this when installing the package.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+*   Indent is 2 spaces
+*   Code should pass the `eslint` linter
+*   Vertical whitespace helps readability, don’t be afraid to use it
+
+## Building
+
+The simplest procedure is to just run `npm run watch` in a terminal. This will
+launch `ucompiler` and tell it to watch the `src` directory, automatically
+recompiling any file that is updated for you. If you prefer to run this step
+manually, running `npm install` will also run `ucompiler go` for you.
```
Comment (steelbrain) [inline CONTRIBUTING.md]: `ucompiler go src` would make sure it never scans any file other than that directory

Comment (Arcanemagus) [inline CONTRIBUTING.md]: Ah, I assumed the `.ucompilerrc` file handled that, will update the docs.

File: package.json [hunk 2]
```diff
@@ -10,22 +10,27 @@
     "atom": ">0.50.0"
   },
   "scripts": {
-    "lint": "eslint ."
+    "lint": "eslint .",
+    "watch": "ucompiler watch src",
+    "prepublish": "ucompiler go"
```
Comment (steelbrain) [inline package.json]: `ucompiler go src`

Comment (steelbrain) [inline package.json]: Also, another script named `compile` would be nice, so if someone wants to compile it real quick they can do `npm run compile`

File: CONTRIBUTING.md (hunk 1 above)
Comment (steelbrain) [inline CONTRIBUTING.md]: It does handle that, but it still has the scan other files to validate their end-file does not match a known glob. I have the enhanement in mind where it'll be able to ignore parent dirs of non-used files

**Comment by steelbrain** [comment]
No idea what's wrong with my `apm install` but it's not allowing any kind-of ES6
```
apm te%                                                                                                                                        ~/.a/p/linter-eslint ❯❯❯ apm install                                                                                 AtomLinter-document-build
Installing modules ✗
> linter-eslint@6.0.0 prepublish /home/steel/.atom/packages/linter-eslint
> ucompiler go src


npm WARN optional dep failed, continuing fsevents@1.0.6

/home/steel/.atom/packages/linter-eslint/node_modules/ucompiler/lib/main.js:43
  let path = arguments.length <= 0 || arguments[0] === undefined ? null : argu
  ^^^
SyntaxError: Unexpected strict mode reserved word
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/steel/.atom/packages/linter-eslint/node_modules/ucompiler/cli/main.js:6:17)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)

npm ERR! Linux 4.2.0-25-generic
npm ERR! argv "/usr/share/atom/resources/app/apm/bin/node" "/usr/share/atom/resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/home/steel/.atom/.apm/.apmrc" "--userconfig" "/home/steel/.atom/.apmrc" "install" "--target=0.34.3" "--arch=x64"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE
npm ERR! linter-eslint@6.0.0 prepublish: `ucompiler go src`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the linter-eslint@6.0.0 prepublish script 'ucompiler go src'.
npm ERR! This is most likely a problem with the linter-eslint package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ucompiler go src
npm ERR! You can get their info via:
npm ERR!     npm owner ls linter-eslint
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/steel/.atom/packages/linter-eslint/npm-debug.log
```

Further investigation shows that it failed on `ucompiler go src`, however doing this works
```
~/.a/p/linter-eslint ❯❯❯ ./node_modules/.bin/ucompiler go src                                                    ⏎ AtomLinter-document-build ✱
  UCompiler:Save Saving src/helpers.js to lib/helpers.js +0ms
  UCompiler:Save Saving src/main.js to lib/main.js +3ms
  UCompiler:Save Saving src/reporter.js to lib/reporter.js +1ms
  UCompiler:Save Saving src/worker-helpers.js to lib/worker-helpers.js +0ms
  UCompiler:Save Saving src/worker.js to lib/worker.js +0ms
```
Even this works
```
~/.a/p/linter-eslint ❯❯❯ ATOM_SHELL_INTERNAL_RUN_AS_NODE=1 /usr/share/atom/atom ./node_modules/.bin/ucompiler go src
  UCompiler:Save Saving src/helpers.js to lib/helpers.js +0ms
  UCompiler:Save Saving src/main.js to lib/main.js +4ms
  UCompiler:Save Saving src/reporter.js to lib/reporter.js +1ms
  UCompiler:Save Saving src/worker-helpers.js to lib/worker-helpers.js +0ms
  UCompiler:Save Saving src/worker.js to lib/worker.js +0ms
```

To repro:
```
rm -rf node_modules
apm install
```
cc @Arcanemagus

**Comment by steelbrain** [comment]
Aaha! APM has it's own bundled version of node that does not match that of Atom
```
~/.a/p/linter-eslint ❯❯❯ /usr/share/atom/resources/app/apm/bin/node --version                                    ⏎ AtomLinter-document-build ✱
v0.10.40
```

**Comment by Arcanemagus** [comment]
@nmote any comments on this before it gets merged as a fresh set of eyes on it?

**Comment by nmote** [comment]
Sure, I'm in the middle of something right now but if you give me an hour I will rebase my autofix stuff on top and try out the workflow.

**Comment by nmote** [comment]
Seems to work. I'll put up a PR shortly.

**Comment by Arcanemagus** [comment]
Merging this then, thanks for the feedback. Hopefully we can automate this...

**Pull request merged**